src/backend/metal/src/command.rs

```diff
@@ -352,6 +367,16 @@ impl State {
                     .chain(iter::once(com_samplers))
                     .chain(com_push_constants)
             });
+        let com_used_resources = self.descriptor_sets
```

Comment:
Why do we always have to do this? Should we only track used resources when arguments buffers are enabled? Or maybe this is part of the fix for #2848?

Reply:
If there are no argument buffers, these lists are empty. At the end of the day, argument buffer should be our preferred path to go, so that's what we optimize for.